### PR TITLE
fix(record): valid assignments in record compact constructors

### PR DESCRIPTION
**Pull request by xzel23**
I have created a second test case for record compact constructors to demostrate the problem. I have created a PR even if I don't have a fix yet to make reproducing easier: either run CtRecordTest.testCompactConstructor2() directly from your IDE (in IntelliJ it's possible to run disabled tests from the IDE, don't know about eclipse), or remove the "@Disabled" annotation from said method.

### CODE REVIEW

**Comment by xzel23** [comment]
Please review. This fixes #4377.

**Comment by slarse** [comment]
@SirYwell can you review this?

@xzel23 Could you please add "fix #4377" to the PR body (first post)? That will automatically close the issue when this is merged. See this line of the [contributing guidelines](https://github.com/INRIA/spoon/blob/master/CONTRIBUTING.md)

> If the pull request resolves an issue, simply add "fix #issueNumber" or "close #issueNumber" to the description, see [doc](https://docs.github.com/en/free-pro-team@latest/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue)

**Review by SirYwell — CHANGES_REQUESTED** [review]
I left two comments for possible improvements. The fix itself looks good.

I also figured out why I wasn't able to reproduce it, as I even checked out your commit before you added the fix. I ran the tests with Java 11 locally, and as JDT can't resolve `java.lang.Record` there, it does not add the assignments... :)

File: src/test/java/spoon/test/record/CtRecordTest.java [hunk 1]
```diff
@@ -8,6 +8,9 @@
 import java.util.Collection;
 import java.util.stream.Collectors;
 import javax.validation.constraints.NotNull;
+
+import org.junit.Ignore;
+import org.junit.jupiter.api.Disabled;
```
[the remark this answers is not in the document]
Comment (xzel23) [reply src/test/java/spoon/test/record/CtRecordTest.java:13]: removed the imports, will update the PR once I'm done with the other comments and unit tests are green

File: src/main/java/spoon/support/compiler/jdt/ContextBuilder.java [hunk 2]
```diff
@@ -115,6 +116,9 @@ void enter(CtElement e, ASTNode node) {
 			// For some element, we throw an UnsupportedOperationException when we call setType().
 		}
 
+		if (e instanceof CtAssignment) {
+			e.setImplicit((node.bits & ASTNode.IsImplicit) != 0);
+		}
```
[the remark this answers is not in the document]
Comment (xzel23) [reply src/main/java/spoon/support/compiler/jdt/ContextBuilder.java:121]: Yes, that's of course much better.

**Review by SirYwell — APPROVED** [review]
Looks good!

**Comment by monperrus** [comment]
Thanks a lot @xzel23 for the fix and @SirYwell for the review.

**Pull request merged**